### ISSUE TITLE
[FEAT] Add an ability to use OpenAI / VoyageAI / Cohere embeddings with SDPMChunker via LiteLLM

### ISSUE THREAD

**Issue by rodion-m** [bhavnicksm/chonkie#38]
We can make something line `EmbeddingsProvider` abstraction with many implementations like `SentenceTransformerEmbeddingsProvider`, `LiteLlmEmbeddingsProvider`, `FastInferenceEmbeddingsProvider` and so on.

**Comment by rodion-m** [comment]
@bhavnicksm could you please try to add this project into Copilot Workspace and try to implement this issue with it? I believe that it should solve it.

**Comment by bhavnicksm** [comment]
Hey @rodion-m !

Thanks for opening an issue 😊

Yes! That's the plan, we have a `BaseEmbeddings` class, that we built to make initial embeddings support for Chonkie 🙌


Some things remaining to plan out is the integration and refactor for Chunkers to use BaseEmbeddings and default Embeddings initialisation scheme...


Planning to continue support for SentenceTransformers as the default but also make it so that a str input inside chunkers could be supported for the users to avoid having too bulky inits on the API side...


Happy to accept comments and PRs for this! 

Thanks

**Comment by rodion-m** [comment]
Yeah, thanks a lot for your work and for a great library. It's great that it's universal and not linked to LangChain / LlamaIndex.

**Comment by bhavnicksm** [comment]
> Yeah, thanks a lot for your work and for a great library. It's great that it's universal and not linked to LangChain / LlamaIndex.

Thank you so much! Appreciate the kind words :)

**Comment by bhavnicksm** [comment]
> @bhavnicksm could you please try to add this project into Copilot Workspace and try to implement this issue with it? I believe that it should solve it.

Yeah, I'll try with Copilot workspace... Till now, I've had a slightly negative experience where it would introduce some bugs that I would need to fix later on. So I'm a bit careful with it, taking only suggestions from copilot and not full PRs.
